### DISCUSSION TITLE
Labels with fn.readers.video?

### DISCUSSION THREAD

**Discussion by hbfreed** [ROCm/MIVisionX#1266] [Q&A] [answered]
Is it possible to actually have labels with rocAL's fn.readers.video? I have tried a bunch of different things, but just can't get labels out. This is my most recent attempt, with lists for filenames and labels (read in from a .csv). The ROCALVideoIterator is ripped pretty much straight from the video example found in the rocAL examples folder. 

I'm using an EPYC 7402p and a 7900xtx on Ubuntu 22.04.3 LTS. 

Here's the simple pipe that I'm using:

```
fileroot = 'Documents/val'
size = len(os.listdir(file_root))

df = pd.read_csv(val_labels.csv,header=None)
filenames_list, labels_list = df[0].tolist(),df[1].tolist()

pipe = Pipeline(batch_size=2, num_threads=48, seed=10, device_id=0,
            rocal_cpu=False, tensor_dtype = types.FLOAT16,
            prefetch_queue_depth = 20)
with pipe:
    videos = fn.readers.video(file_root=file_root,
                              shard_id=0, 
                              num_shards=1, 
                              random_shuffle=True,
                              sequence_length=130,
                              filenames=filenames_list,
                              labels=labels_list)
pipe.set_outputs(videos)



pipe.build()

train_loader = ROCALVideoIterator(pipe,size)
for i, (inputs,labels) in enumerate(train_loader):
    print(inputs.shape,labels.shape)
```

**Comment by kiritigowda** [comment]
@LakshmiKumar23 & @rrawther can you help with this issue?

**Reply by rrawther** [reply]
Hi Kiriti, I have asked the developers to look into this. But it will be supported in the new rocAL repo

**Comment by kiritigowda** [comment, marked as answer]
Hi @hbfreed rocAL is now an independent component and the repo is public now - https://github.com/ROCm/rocAL

There is an open PR to add support for labels - https://github.com/ROCm/rocAL/pull/89

**Comment by hbfreed** [comment]
Thank you both very much! Closing this now, that PR looks great. Excited to put it to good use :) .

**Comment by kiritigowda** [comment]
@hbfreed - Please let us know if you run into any issue. Thanks!
